app/controllers/Sheets.scala

```
@@ -44,16 +46,34 @@ class Sheets(db: Database, aab: AuthActionBuilder, cc: ControllerComponents)(imp
 
   def getRows(pId: Int, sId: Int) = AttemptAction { req =>
     // Can't really serialize a map of Int -> T so we use this version of coverage while we're folding
-    case class MappedCoverage(var hits: Int, var conflicts: Int, var parseFailures: Int)
+    case class MappedCoverage(var hits: Int, var misses: Int, var conflicts: Int, var parseFailures: Int)
 
-    db.getRows(sId).map(rows => {
+		val filters = req.queryString.getOrElse("filter[]", Nil).map(filter => {
+			val parts = filter.split(":")
+
+			Try {
```

Should we return `BadRequest` for parse failures rather than a default value?

---

Mmmm sounds reasonable. I'm not sure how to communicate to a user that the filters are invalid. Maybe just suggest they refresh?

---

Yeah it was more for server-side cleanliness, I think it's fine for the UI just to barf